### PR TITLE
*: Support the statement of `ALTER [COLUMN] col_name SET DEFAULT`

### LINKED ISSUE #2383: Support the statement of `ALTER [COLUMN] col_name SET DEFAULT `
This method can change the default value.
See [URL](http://dev.mysql.com/doc/refman/5.7/en/alter-table.html).

### PR DESCRIPTION
**Pull request by zimulala**
Fix #2383

### CODE REVIEW

File: parser/parser.y [hunk 1]
```diff
@@ -880,6 +880,17 @@ AlterTableSpec:
 			NewColumn: 	$4.(*ast.ColumnDef),
 		}
 	}
+|	"ALTER" "COLUMN" ColumnName "SET" "DEFAULT" SignedLiteral
+	{
+		option := &ast.ColumnOption{Expr: $6.(ast.ExprNode)}
```
Comment (coocood) [inline parser/parser.y]: Set ColumnOption Type.

File: ddl/ddl_api.go [hunk 2]
```diff
@@ -943,6 +945,54 @@ func (d *ddl) ModifyColumn(ctx context.Context, ident ast.Ident, spec *ast.Alter
...
+	}
+	col.DefaultValue = value
+	err = checkDefaultValue(col, true)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	// Check column default value.
+	// TODO: Put this code to checkDefaultValue and fix issue 2600.
+	colInfo := col.ToInfo()
+	_, _, err = table.GetColDefaultValue(ctx, colInfo)
```
Comment (coocood) [inline ddl/ddl_api.go]: Why try to get default value here?

File: parser/parser.y (hunk 1 above)
Comment (zimulala) [inline parser/parser.y]: We can use original type information.

File: ddl/ddl_api.go (hunk 2 above)
Comment (zimulala) [inline ddl/ddl_api.go]: It uses to check column default value as the comment say.

**Comment by XuHuaiyu** [comment]
LGTM

**Comment by zimulala** [comment]
PTAL @coocood

**Comment by coocood** [comment]
LGTM

**Pull request merged**